scripts/agent_mkconfig.py

```diff
@@ -98,12 +98,21 @@ def __init__(self):
         # cores and devices. We need to ascend the tree to find the container.
         parent_map = {child: parent for parent in self._tree.iter() for child in parent}
         for i in range(len(self._nodes)):
-            if int(self._nodes[i].get("os_index", 0)) != i:
+            node = self._nodes[i]
+            if int(node.get("os_index", 0)) != i:
                 raise RuntimeError("NUMA nodes are not numbered contiguously by the OS")
-            while not self._nodes[i].findall(".//object[@type='PU']") or not self._nodes[i].findall(
-                ".//object[@type='Bridge']"
+            while not node.findall(".//object[@type='PU']"):
```

Comment:
Maybe it's just the length of the fully-stitched `node.final(...)` call that makes me uneasy, but
* What do you think about a small helper function to make it look a bit neater?

Likely wholly unnecessary as a change; I just don't like the look of the `findall` arg - so aggressive!

Reply:
Yeah, it was bugging me slightly too. Will make the change.